pkg/master/master.go

```diff
@@ -285,6 +288,16 @@ func setDefaults(c *Config) {
 		glog.V(4).Infof("Setting master service IP to %q (read-write).", serviceReadWriteIP)
 		c.ServiceReadWriteIP = serviceReadWriteIP
 	}
+	if c.ServicePorts == nil {
```

Comment:
And a test case (even a simple one)

On Oct 2, 2015, at 3:41 PM, Tim Hockin <notifications@github.com> wrote:

In pkg/master/master.go
<https://github.com/kubernetes/kubernetes/pull/14871#discussion_r41059515>:

> @@ -285,6 +288,16 @@ func setDefaults(c *Config) {
>  		glog.V(4).Infof("Setting master service IP to %q (read-write).", serviceReadWriteIP)
>  		c.ServiceReadWriteIP = serviceReadWriteIP
>  	}
> +	if c.ServicePorts == nil {

comment somewhere about WHY this is injectable - it's not at all obvious

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/14871/files#r41059515>.